api/cookie.go

```diff
@@ -0,0 +1,73 @@
+package api
+
+import (
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gorilla/securecookie"
+)
+
+const (
+	// CookieName is the name of the cookie where we store the user's JWT token.
+	CookieName = "skynet-jwt"
+	// cookieValidity specifies for how long the cookie is valid. In seconds.
+	cookieValidity = 604800 // one week
+
+	// envCookieDomain holds the name of the environment variable for the
+	// domain name of the portal
+	envCookieDomain = "COOKIE_DOMAIN"
+	// envCookieHashKey holds the name of the env var which holds the key we use
+	// to hash cookies.
+	envCookieHashKey = "COOKIE_HASH_KEY"
+	// envCookieEncKey holds the name of the env var which holds the key we use
+	// to encrypt cookies.
+	envCookieEncKey = "COOKIE_ENC_KEY"
+)
+
+var (
+	// _sc holds an instance of `securecookie`, so we don't instantiate it more
+	// than once.
+	_sc *securecookie.SecureCookie = nil
```

Comment:
A more idiomatic solution would be

```go
sc = func() *securecookie.SecureCookie {
	var hashKey = []byte(os.Getenv(envCookieHashKey))
	var blockKey = []byte(os.Getenv(envCookieEncKey))
	return securecookie.New(hashKey, blockKey)
}()
```

or alternatively to use the `init` function. https://golang.org/doc/effective_go.html#init
Then you don't need the underscore prefix to indicate a special variable.

Reply:
I thought about the `init` approach but I've never heard anything good about it and multiple sources complain about the added "magic" and weird bugs they managed to create with it. I'd rather stay away from it where possible and the proposed function solution is pretty nice and readable.